### DISCUSSION TITLE
Do command aliases not appear in the help by design?

### DISCUSSION THREAD

**Discussion by lppedd** [ajalt/clikt#470] [Q&A] [answered]
Example:
```kotlin
override fun aliases(): Map<String, List<String>> =
  mapOf(
    "ul" to listOf("update-licenses"),
  )
```

In that case I'd expect `ul` to be shown beside `update-licenses`.  
Otherwise how does a user know there is an alias available?

Although, I think the aliases design might have to change to accomodate this use case.

**Comment by ajalt** [comment, marked as answer]
You're right that aliases aren't part of the [data passed to the help formatter][1], so there isn't currently any way to include them in the help automatically. I'm even convinced that adding them automatically is a good idea, since they can be defined dynamically e.g. the way git aliases work.

I would suggest manually adding the info you want to your command's help or its epilog.

[1]: https://github.com/ajalt/clikt/blob/b87775d354633e9e55e58ebc0cb57cb55ab09cb0/clikt/src/commonMain/kotlin/com/github/ajalt/clikt/output/HelpFormatter.kt#L63
